compiler/rustc_middle/src/ty/context.rs

```diff
@@ -130,6 +130,42 @@ impl<'tcx> Interner for TyCtxt<'tcx> {
     ) -> (Self::Ty, ty::Mutability) {
         (ty, mutbl)
     }
+
+    fn mk_canonical_var_infos(
+        &self,
+        infos: &[rustc_type_ir::CanonicalVarInfo<Self>],
+    ) -> Self::CanonicalVars {
+        (*self).mk_canonical_var_infos(infos)
+    }
+
+    fn mk_bound_ty(
+        &self,
+        debruijn: rustc_type_ir::DebruijnIndex,
+        var: rustc_type_ir::BoundVar,
+    ) -> Self::Ty {
+        Ty::new_bound(*self, debruijn, ty::BoundTy { var, kind: ty::BoundTyKind::Anon })
+    }
+
+    fn mk_bound_region(
+        &self,
+        debruijn: rustc_type_ir::DebruijnIndex,
+        var: rustc_type_ir::BoundVar,
+    ) -> Self::Region {
+        Region::new_bound(
+            *self,
+            debruijn,
+            ty::BoundRegion { var, kind: ty::BoundRegionKind::BrAnon },
+        )
+    }
+
+    fn mk_bound_const(
+        &self,
```

Comment:
same with these functions, take `self` by value and maybe require copy on interner? :thinking: if r-a does not have a wrapper it can impl interner for `&TheirInterner`